designate/storage/sqlalchemy/__init__.py

```diff
@@ -1436,6 +1436,12 @@ def create_pool(self, context, pool):
         :param context: RPC Context.
         :param pool: Pool object with the values to be created.
         """
+        if not context.is_admin:
+            if context.project_domain_id != pool.domain_id and not pool.shared:
```

Comment:
Do we really need `and not pool.shared: `?
Because from the start Pool is not created as Shared, as per https://github.com/sapcc/designate/pull/52/files#diff-f26d6552896103941b20da10d5cde8879bca607b3ded9d90c777fa7bece1f1bdR88

Reply:
Updated